File: MyFinance.Frontend/src/types/InvestmentType.ts
/**
 * Classes de ativos de investimento.
 * Corresponde ao enum InvestmentType.cs no backend (serializado como número).
 */
export const InvestmentType = {
    RendaFixa: 1,
    Acao: 2,
    FII: 3,
    Cripto: 4,
    ETF: 5,
} as const;

export type InvestmentType = (typeof InvestmentType)[keyof typeof InvestmentType];

File: MyFinance.Domain/Enums/InvestmentType.cs
namespace MyFinance.Domain.Enums;

/// <summary>
/// Define as classes de ativos suportadas pela gestão de investimentos.
/// </summary>
public enum InvestmentType
{
    /// <summary>
    /// Renda Fixa (ex: Tesouro Direto, CDB, LCI/LCA).
    /// </summary>
    RendaFixa = 1,

    /// <summary>
    /// Ações negociadas em bolsa (ex: PETR4, VALE3).
    /// </summary>
    Acao = 2,

    /// <summary>
    /// Fundos de Investimento Imobiliário (ex: HGLG11, MXRF11).
    /// </summary>
    FII = 3,

    /// <summary>
    /// Criptoativos (ex: Bitcoin, Ethereum).
    /// </summary>
    Cripto = 4,

    /// <summary>
    /// Exchange Traded Funds (ex: BOVA11, IVVB11).
    /// </summary>
    ETF = 5
}
